ci/check-protobuf-stability.sh

```diff
@@ -35,42 +51,76 @@ function check_non_lf_protos() {
+    # This in turns requires that we pass the `--against-config` flag for any check that is run on versions > 1.17
+    if [[ $BUF_CONFIG_UPDATED = true ]]
+    then
+      buf breaking --config "${buf_module}" --against "$BUF_GIT_TARGET_TO_CHECK" --against-config "${buf_module}"
+    else
+      buf breaking --config "${buf_module}" --against "$BUF_GIT_TARGET_TO_CHECK"
+    fi
+  done
 
-  # For `main` and PRs targeting `main`, we simply check against the most recent
-  # stable tag.
-  #
-  # For PRs targeting release branches, we should really check against
-  # all the most recent stable tags reachable from either the current branch or
-  # from previous release branches (say, against both `1.17.1` and `1.16.2`
-  # created after the `release/1.17.x` branch).
-  # Instead, we approximate by checking only against the most recent stable tag
-  # reachable from the current branch, under the assumption that if a lesser
-  # release branch contains a protobuf change, then it will also be present in
-  # higher ones either through a shared commit or a back-port from `main`.
+}
+
+case "${1:---stable}" in
+-h | --help)
+  cat <<USAGE
+Usage: ./check-protobuf-stability.sh [options]
+
+Options:
+  -h, --help: shows this help
+  --stable:     check against the latest stable tag (default)
+  --target:     check against the tip of the target branch
+USAGE
+  exit
+  ;;
+--stable)
+  # Check against the most recent stable tag.
   #
-  # Finally, this check does not need to run on release branch commits because
+  # This check does not need to run on release branch commits because
   # they are built sequentially, so no conflicts are possible and the per-PR
   # check is enough.
+  readonly RELEASE_BRANCH_REGEX="^release/.*"
   GIT_TAG_SCOPE=""
+  # For PRs targeting release branches, we should really check against
+  # all the most recent stable tags reachable from either the current branch or
+  # from previous release branches (say, against both 1.17.1 and 1.16.2
+  # created after the release/1.17.x branch).
+  #
+  # Check against the most recent stable (non-snapshot) tag reachable from the current branch,
+  # under the assumption that if a previous release branch contains a protobuf change,
+  # then it will also be present in former ones (previous/former with regards to versioning).
   if [[ "${TARGET}" =~ ${RELEASE_BRANCH_REGEX} ]]; then
     GIT_TAG_SCOPE="--merged"
   fi
-
   readonly LATEST_STABLE_TAG="$(git tag ${GIT_TAG_SCOPE} | grep -v "snapshot" | sort -V | tail -1)"
-  echo "Checking protobufs against tag '${LATEST_STABLE_TAG}'"
-  for buf_module in "${BUF_MODULES_AGAINST_STABLE[@]}"; do
-    (eval "$(dev-env/bin/dade assist)" ; buf breaking --config "${buf_module}" --against ".git#tag=${LATEST_STABLE_TAG}")
-  done
-
-}
+  # The current stable release is v1.17 and it includes the buf config file with the default name `buf.yml`.
+  # Once we have the v1.18 release we can remove this check for the updated config (KVL-1131)
```

Comment:
```suggestion
  # v1.17 includes the buf config file with the default name `buf.yml`.
```
Time references in code become stale very fast.

Reply:
How do you mean by time references? Do you refer to the version? 
In this case, the version is actually absolute, I don't see a way how this comment can become stale, as we would only be removing it when we advance to the 1.18 release.

Reply:
Updated the comment, I hope it's clearer now. 👍 for your points